### ISSUE TITLE
Inconsistencies showing date in xml, default display mode, and full display mode

### ISSUE THREAD

**Issue by jodygarnett** [metadata101/iso19139.ca.HNAP#135]
Sample data shows inconsistencies between date displayed in `xml`, `default` display mode, and `full` display mode.

*Ocean Salmon Program* `b16db0fe-a4ec-4844-819e-630e33137b6d`
* dateStamp: `2018-12-17T12:41:20` (displayed full `Mon Dec 17 2018 12:41:20 GMT-0800`)
* created: `1993` (display full: `1992` default `1992-12-31` )
* published: `2018` (display full `2017`)
* revision: `2018-12` (display full: `Dec 2018`)
* begin: `1990-07-25` (display full: `25 Jul 1990`)
* end: `1993-07-27` (display full: `27 Jul 1993`)

*Harbour Porpoise Presence, Maritimes Region* `58ea48ab-f052-48ab-9c18-4353e51b8bea`

* dateStamp: `2018-09-21T12:08:41` (displayed full `Fri Sep 21 2018 12:08:41 GMT-0700`)
* created: `2016-11-18` (displayed full: `18 Nov 2016` default `2016-11-18`)
* published: `2018-06-21` (displayed full `21 Jun 2018`)
* revision
* begin: `2018-05-09` (displayed full `09 May 2018`)

*Sites d'échouerie potentiels* `3b0ccbbe-e0f7-4e6f-a6aa-8b327c69a169`

* dateStamp: `2020-03-25T14:12:19` (displayed `Wed Mar 25 2020 14:12:19 GMT-0700`)
* creation: `2001` (display default: `2000-12-31` full:`2000`)
* publication: `2018` (display full: `2017`)
* revision:
* begin: `1978` (displayed full `1978`)
* end: `2000` (displayed full `2000` )

**Comment by ianwallen** [comment]
I suspect that some of the issues are related to timezone.

i.e. 
publication: 2018 (display full: 2017)

2018 in the metadata record.
2018  -0700  due to your timezone give 2017-12-31-17:00 when then gets truncated to 2017

Not sure how easy this will be to fix.
Suggestion - Since all dates are being indexes as GMT - I would suggest that all times displayed are also in GMT (not the users timezone)

But the problem that when a use enter 2018-12-17T12:41:20 - they are probably talking about their local timezone and not GMT. 
So anytime we ask for a users to enter time, we would need to ensure that a timezone is associated to that time. This can be done by prompting the user to specify the timezone of the time (maybe defaulting to their timezone?)

Not sure if the ISO 19139 specification contains a timezone field to indicate the default timezone for all times added in the record?

**Comment by jodygarnett** [comment]
Yes it is very much related to timezone, @juanluisrp has been working on ISODate fixes in other branch, so I expect this will be resolved over time (no pun intended). 

I would like to keep this ticket open to use for testing any ISODate fix, and wanted to make a note of how confusing this was for evaluating https://github.com/metadata101/iso19139.ca.HNAP/pull/134 fix.

Aside: I should note that for the most part things are okay, it is mostly when users enters a date `2014-02-01`, or a month `2016-13` or even worse a year `2007` that we run into trouble. I kind of feel that in these cases we should use "AOE" (any where on earth) time range for the given day to better reflect the intension.

**Comment by fxprunayre** [comment]
This would be solved by https://github.com/geonetwork/core-geonetwork/pull/5061 and https://github.com/geonetwork/core-geonetwork/issues/1970. A workaround (for the JS app) for now is to shift date to the server timezone UtilityDirective.js.

**Comment by ianwallen** [comment]
@jodygarnett 
I think gco:DateTime supports dates with timzeones 

https://www.wmo.int/schemas/wmdr/1.0RC4/documentation/schemadoc/schemas/basicTypes_xsd_1/elements/DateTime.html

as it is based on xs:dateTime

https://www.w3schools.com/xml/schema_dtypes_date.asp

So it seems that `2018-09-21T12:08:41Z`  would also be a good date to add to your test.

It would also be nice if the time picker supported specifying the timezone as it would be clearer as to what the real time is.  Personally I don't think any time should exists without the timezone specified.

**Comment by jodygarnett** [comment]
Actually looking at the dateStamp `2018-12-17T12:41:20` it seems correct, as my server is running in PST and this shows up as with GMT-0700.

**Comment by juanluisrp** [comment]
> It would also be nice if the time picker supported specifying the timezone as it would be clearer as to what the real time is. Personally I don't think any time should exists without the timezone specified.

This is part of https://github.com/geonetwork/core-geonetwork/pull/5061 PR. A timezone picker has been added there in the date/time fields of the editor.

**Comment by ianwallen** [comment]
@juanluisrp, I see that PR is targeting the 4.X branch.  Are there plans to backport these changes to the 3.X/master branch once completed?  Also if backporting,  do you have any estimates as to when this work is expected to be complete?

Thank you for the great contributions - we do see lots of time related issues so we are anxious to see these changes get merged and backported.

**Comment by jodygarnett** [comment]
@ianwallen based on recent discussion no group has stepped up with resources to backport to the 3.x branch.
